label_studio/data_export/api.py

```diff
@@ -436,23 +442,109 @@ def get_queryset(self):
+            # The below header tells NGINX to catch it and serve, see docker-config/nginx-app.conf
+            redirect = '/file_download/' + protocol + '/' + url.replace(protocol + '://', '')
+
+            response['X-Accel-Redirect'] = redirect
+            response['Content-Disposition'] = 'attachment; filename="{}"'.format(file.name)
+            return response
         else:
-            file_ = instance.convert_file(export_type)
-        
-        if file_ is None:
-            return HttpResponse("Can't get file", status=404)
+            if export_type is None:
+                file_ = snapshot.file
+            else:
+                file_ = snapshot.convert_file(export_type)
 
-        ext = file_.name.split('.')[-1]
+            if file_ is None:
+                return HttpResponse("Can't get file", status=404)
 
-        response = RangedFileResponse(request, file_, content_type=f'application/{ext}')
-        response['Content-Disposition'] = f'attachment; filename="{file_.name}"'
-        response['filename'] = file_.name
-        return response
+            ext = file_.name.split('.')[-1]
+
+            response = RangedFileResponse(request, file_, content_type=f'application/{ext}')
+            response['Content-Disposition'] = f'attachment; filename="{file_.name}"'
+            response['filename'] = file_.name
+            return response
+
+
+def async_convert(converted_format_id, export_type, **kwargs):
+    with transaction.atomic():
+        try:
+            converted_format = ConvertedFormat.objects.get(id=converted_format_id)
+        except ConvertedFormat.DoesNotExist:
+            logger.error(f'ConvertedFormat with id {converted_format_id} not found, conversion failed')
+            return
+        if converted_format.status != ConvertedFormat.Status.CREATED:
+            logger.error(f'Converson for export id {snapshot_id} to {export_type} already started')
+            return
+        converted_format.status = ConvertedFormat.Status.IN_PROGRESS
+        converted_format.save(update_fields=['status'])
+
+    snapshot = converted_format.export
+    project = snapshot.project
+    converted_file = snapshot.convert_file(export_type)
+    md5 = Export.eval_md5(converted_file)
+
+    now = datetime.now()
+    file_name = f'project-{project.id}-at-{now.strftime("%Y-%m-%d-%H-%M")}-{md5[0:8]}.{export_type.lower()}'
+    file_path = (
+        f'{project.id}/{file_name}'
+    )  # finally file will be in settings.DELAYED_EXPORT_DIR/project.id/file_name
+    file_ = File(converted_file, name=file_path)
+    converted_format.file.save(file_path, file_)
+    converted_format.status = ConvertedFormat.Status.COMPLETED
+    converted_format.save(update_fields=['file', 'status'])
+
+
+def set_convert_background_failure(job, connection, type, value, traceback):
+    from data_export.models import ConvertedFormat
+
+    convert_id = job.args[0]
+    ConvertedFormat.objects.filter(id=convert_id).update(status=Export.Status.FAILED)
```

Comment:
let's add traceback to ConvertedFormat and save it too, it will be very helpful for further usage.

Reply:
Added